### DISCUSSION TITLE
fog of war for outfits

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#7768] [Ideas]
Greetings,

I find it weird that I scan a ship of a race I never encountered for outfits and see all the outfits and what they are.
would implementing a fog of war like feature is something interesting?
a new outfit is known when either one enters an outfitter or actually board a ship.
without that scans will show "unknown engine/turret/gun/.../outfit"

thoughts?

**Comment by Hecter94** [comment]
There's been talk of only showing the category of installed outfits until you've done "something" to give you a better knowledge of them. 

As in, you'd scan an unknown ship and see that it has 
X Guns
X Turrets
X Systems
X Power
etc.

With that said, I feel like scanning is already pretty underutilized, so I'd be hesitant to make it even less useful.

**Reply by bene-dictator** [reply]
It could be that scanning reveals the name / stats of the outfit. If you haven't scanned the ship, but you've boarded it, you have no reason to know what is what, only that the outfit does something. Taking home an alien reactor for study is more interesting than finding out immediately that the reactor is one of those Korath ones that overheat your ship in seconds, and could get you 5 million credits because... it's just worth that much? Does the player have someone on board to value alien outfits on the fly? You can't even buy Korath fusion reactors, and I wouldn't buy them for 25 million anyway... But that's for another discussion.

**Reply by Zitchas** [reply]
Actually, if you are in the Remnant storyline, they actually mention how they use QEC communicators to upload identifying data from your sensors to them, and they in return provide you with names and basic info. Effectively, there's an in-lore explanation for how you have this information.... But only once you are working for the Remnant. For the rest of the game... Yeah, it's just a giant plot-hole as to how the player knows this stuff.

Probably falls into the same boat as:
- How come we can loot an engine off some alien from another galaxy (Wanderer, for instance), bring it back to an outfitter anywhere else in the galaxy, and the mechanics there will be able to understand it well enough to install it into our ship, hook up power and everything else, and get it up and working at 100% in less than a day. 
- How our pilot can board an automated alien warship and be able to hack it to the point of having it have full functionality but consider the pilot to be it's ultimate leader/owner to be obeyed unquestiongly?
- How everyone's equipment from every race in the galaxy all seems to operate on the exact same modular inter-changeable ship construction system? Any outfit can be taken out of any ship and installed into any other?

**Comment by Zitchas** [comment]
I would like to have more research capabilities that we could use to encourage a player to actually study opponents and learn more about things - I'd definitely make use of those capabilities in story. It would certainly make sense.

**Comment by daggs1** [comment]
so bottom line, is it interesting enough?

**Reply by vitalchip** [reply]
I think the idea is sound, as things stand the player knows too much, too soon. Taking a step backwards in this or any other aspect will not always be well received but if the intention is to create another area of achievement and to create the situation where 'experience counts' then it's worth it.

Scanners and sensors are a bit lacklustre in general, underused and underuseful. Why does anyone buy a cargo scanner?  1. A mission makes you buy it. 2. See 1.

The range is too short and the method of capturing the data is too clumsy. I'd rather enter a system, turn on active scanning and let it get the results as I fly around, then look at them when required.

Running active scanning could make your ship more easily and precisely targeted, give missiles a better lock and generally put your head above the parapet, so maybe not something you want to just leave switched on all the time.

The same sort of logic applies to ships, why does the player know every model of ship?  Is there to be no mystery!

Transponders are a logical answer and certainly commercial shipping would likely have them but why are the Remnant flying around transponding to all? Why are the Kor Drones?   

Anyway, on a more practical note.. Perhaps a decent start would be to move the meat of MapOutfitterPanel::Init() and MapShipyardPanel::Init() into PlayerInfo so it can be called by scanning as well as those panels, that way anything the player has seen in an outfitter or a shipyard can be identified and those the player hasn't seen just get a generic category.

We could then push into the set of seen outfits/ships on other events, such as looting an outfit or just boarding the ship. Might need to be wary of cross contamination though, not sure we want the outfitter map to show where we can buy something just because we've looted one on our travels.  Hmm..  some thought needed on that score. Could change it to a map maybe, add something to indicate 'how' it was seen.

Just thought that might even lead in the direction of something missions could use?  Offer if the player is aware of "Electron Beams" for example.

**Reply by daggs1** [reply]
that sound interesting but for now I want to keep the change to min. I have a pause in the game until I figure out the heck I'm doing with the  Emerald Sword, I'll implement a basic idea and we will see from there.

thanks for the feedback
